libV2/tasks/IngestShard.js

```diff
@@ -0,0 +1,54 @@
+const assert = require('assert');
+const BaseTask = require('./BaseTask');
+const { UtapiMetric } = require('../models');
+const config = require('../config');
+const { LoggerContext, shardFromTimestamp } = require('..//utils');
+const { shardIngestLagSecs } = require('../constants');
+
+const logger = new LoggerContext({
+    module: 'IngestShard',
+});
+
+class IngestShardTask extends BaseTask {
+    constructor(...options) {
+        super(...options);
+        this._defaultSchedule = config.ingestionSchedule;
+    }
+
+    async _execute(timestamp) {
+        const endShard = shardFromTimestamp(timestamp - (shardIngestLagSecs * 1000000));
+        logger.debug('ingesting shards', { endShard });
+
+        const available = await this._cache.getShards();
+        const toIngest = available
+            .map(key => key.split(':')[2])
+            .filter(key => parseInt(key, 10) <= endShard);
+
+        if (!toIngest.length) {
```

Comment:
I'd rather avoid using boolean operators on non-booleans (even though some C programmers like to do it :laughing: ), doing an explicit comparison against 0 is more readable.

Note that javascript allows to check for object existence like `if (obj)` or `if (!obj)`, this is a different use that seems more appropriate since an object exists or not (boolean).